homeassistant/components/xiaomi_ble/sensor.py

```diff
@@ -0,0 +1,149 @@
+"""Support for xiaomi ble sensors."""
+from __future__ import annotations
+
+from typing import Optional, Union
+
+from xiaomi_ble import DeviceClass, DeviceKey, SensorDeviceInfo, SensorUpdate, Units
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothCoordinatorEntity,
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothDataUpdateCoordinator,
+    PassiveBluetoothEntityKey,
+)
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    PERCENTAGE,
+    PRESSURE_MBAR,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+SENSOR_DESCRIPTIONS = {
```

Comment:
I assume the integration now picks up all sensors that send temperature, humidity, pressure and battery. This might add all sensors, as almost all of them have battery messages. Does this mean that these sensors are added with only a battery entity? Not a problem, I guess, but we can also decide to comment out the sensors that aren't fully supported yet in the pypi package.

Reply:
Pressure isnt used yet, but yes. I've left all the parsers active and able to capture fields that aren't turned into entities so we can do a second pass and capture them.

The can ignore any devices they don't want to add in the UI, so i'd slightly prefer to leave as is I think.

Reply:
I’m just a bit afraid of all the issues that are going to be created, when people notice that their device only shows a battery entity. There is also one device that has added a counter in the battery data. Like every tenth advertisement, it will send a counter (as battery data) that is increased with one. I had some filter in BLE monitor for that (not in the parser, but in sensor.py. It will look it up, which one it was…)

Reply:
Eh, I'm less worried about that. It's been that way with homekit_controller for a few years.

We can add some boilerplate to the docs about what device classes are support and/or a whitelist of devices we expect to work.

(I actually have a dashboard just for batteries, so I'm totally up for battery only sensors).

Reply:
The CGPR1 is the device with the odd battery sensor. I made a workaround in BLE monitor by storing the last 5 readings in a list, and only process the battery reading if it is in the list. 

https://github.com/custom-components/ble_monitor/blob/de09d4396f2d9ce8f4d44c10c6ef90ebf6483c90/custom_components/ble_monitor/sensor.py#L259

The counter is actually counting downwards, this is what you get without the workaround. 

![image](https://user-images.githubusercontent.com/24375847/180435841-6d59ca14-e25d-4fd3-b4cf-c151a3d2d59f.png)

Reply:
I've moved this into https://github.com/Bluetooth-Devices/xiaomi-ble/issues/3 and will handle it seperately to this PR.